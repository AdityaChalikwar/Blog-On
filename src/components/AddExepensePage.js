import React from 'react'
import ExpenseForm from './ExpenseForm'
import {connect} from 'react-redux'
import {addExpense} from '../actions/expenses'

export class AddExpensePage extends React.Component {
    constructor(props){
        super(props)
    }

    onSubmit = (expense) => {
        this.props.addExpenseFunction(expense)
        this.props.history.push('/')
    }

    render(){
        return (
            <div>
                <h3>Add Expense</h3>
                <ExpenseForm onSubmit={this.onSubmit}/>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    addExpenseFunction: (expense) => dispatch(addExpense(expense))
})

export default connect(undefined, mapDispatchToProps)(AddExpensePage)