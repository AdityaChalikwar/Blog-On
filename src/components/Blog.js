import React from 'react'
import { firebase } from '../firebase/firebase'
import moment from 'moment'
import uuid from 'uuid'

export default class Blog extends React.Component{
    constructor(props){
        super(props)
        this.state = {
            title: props.blog ? props.blog.title : '',
            body: props.blog ? props.blog.body : ''
        }
    }

    onBodyChange = (e) => {
        const body = e.target.value
        this.setState(() => ({body}))
    }

    onTitleChange = (e) => {
        const title = e.target.value
        this.setState(() => ({title}))
    }

    onSubmit = (e) => {
        e.preventDefault()
        this.props.onSubmit({
            title: this.state.title,
            body: this.state.body,
            createdAt: moment().valueOf(), 
            reads: {
                id: firebase.auth().currentUser.uid
            }
        })
    }

    render(){
        return (
            <div className="container">
                <form className="form" onSubmit={this.onSubmit}>
                    <input
                        type='text' 
                        className="text-input"
                        placeholder='Title'
                        autoFocus 
                        value={this.state.title}
                        onChange={this.onTitleChange}
                        required/>
                    <textarea 
                        className="textarea" 
                        placeholder='Start Writing Here........'
                        value={this.state.body}
                        onChange={this.onBodyChange}
                        required/>
                    <div>
                        <button className="button">Save Blog</button>
                    </div>
                </form>
            </div>
        )
    }
}