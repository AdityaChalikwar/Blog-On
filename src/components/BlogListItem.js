import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'
import { connect } from 'react-redux'
import { startSetReads } from '../actions/blogs'
import { firebase } from '../firebase/firebase'

class BlogListItem extends React.Component{
    constructor(props){
        super(props)
    }
    addReads = () => {
        if(Object.values(this.props.reads).indexOf(firebase.auth().currentUser.uid) <= -1)
            this.props.startSetReads(this.props.id, firebase.auth().currentUser.uid)
    }

    render(){
        return (
            <Link 
                to={`/read/${this.props.id}`} 
                onClick={this.addReads} 
                className="list-item" 
                target="_blank">
                <div className="list-item__data">
                    <h4 className="list-item__title">{this.props.title}</h4>
                    <p className="list-item__body">{this.props.body}</p>
                    <span className="list-item__subtitle">Published on {
                        moment(this.props.createdAt).format('MMM Do, YYYY. h:mm A')}</span>
                </div>
                <h4 className="show-for-mobile">Reads : {Object.keys(this.props.reads).length}</h4>
                <h4 className="show-for-desktop">{Object.keys(this.props.reads).length}</h4>
            </Link>
        )
    }
}

const matchDispatchToProps = (dispatch) => ({
    startSetReads: (id, uid) => dispatch(startSetReads(id, uid))
})

export default connect(undefined, matchDispatchToProps)(BlogListItem)