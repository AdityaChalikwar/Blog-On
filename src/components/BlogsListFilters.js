import React from 'react'
import { setTextFilter, sortByDate, sortByReads } from '../actions/filters'
import { connect } from 'react-redux'

class BlogListFilters extends React.Component{
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }

    onChange = (e) => {
        this.props.sortBy(e.target.value)
    }

    render(){
        return (
            <div className="filter-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input 
                            className="text-input"
                            type='text'
                            placeholder='Search Blogs' 
                            value={this.props.filters.text} 
                            onChange={this.onTextChange}/>
                    </div>
                    <div className="input-group__item">
                        <select
                            className="select" 
                            value={this.props.filters.sortBy} 
                            onChange={this.onChange}>
                            <option value='date'>Newest First</option>
                            <option value='reads'>Reads</option>
                        </select>
                    </div>
                </div>
            </div>
        )
    }
}

const matchStateToProps = (state) => ({
    filters: state.filters
})

const matchDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortBy: (filter) => filter === 'date' ? dispatch(sortByDate()) : dispatch(sortByReads())
})

export default connect(matchStateToProps, matchDispatchToProps)(BlogListFilters)