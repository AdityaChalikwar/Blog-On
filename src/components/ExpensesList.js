import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from './ExpenseListItem'
import selectExpenses from '../selectors/expenses'
import expensesTotal from '../selectors/expenses-total'

export const ExpensesList = (props) => {
    const expenseWord = props.expenses.length === 1 ? 'exepense' : 'expenses'
    return (
        <div>
        {
            props.expenses.length === 0 ? (
                <p>No Matches To Filters</p>
            ) : (
                <div>
                <p>Viewing {props.expenses.length} {expenseWord} totalling 
                {
                    props.total.toLocaleString('en-IN', {
                    style: 'currency',
                    currency: 'INR'
                })}</p>
                {props.expenses.map((expense, index) => <ExpenseListItem key={index} {...expense}/>)}
                </div>
            )
        }
        </div>
    )
}

const mapStateToProps = (state) => {
    const expenses = selectExpenses(state.expenses, state.filters)
    return {
        expenses,
        total: expensesTotal(expenses)
    }
}

export default connect(mapStateToProps)(ExpensesList)