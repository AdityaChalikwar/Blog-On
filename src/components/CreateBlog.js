import React from 'react'
import Blog from './Blog'
import { startAddBlog } from '../actions/blogs'
import { connect } from 'react-redux'

export class CreateBlog extends React.Component {
    onSubmit = (blog) => {
        this.props.startAddBlog(blog)
        this.props.history.push('/blogs')
    }
    
    render(){
        return (
            <div>
                <div className="page-header">
                        <h1 className="page-header__title">New Blog</h1>
                </div>
                <Blog onSubmit={this.onSubmit}/>
            </div>
        )
    }
}

const matchDispatchToProps = (dispatch) => ({
    startAddBlog: (blog) => dispatch(startAddBlog(blog))
})

export default connect(undefined, matchDispatchToProps)(CreateBlog)