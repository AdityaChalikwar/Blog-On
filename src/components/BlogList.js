import React from 'react'
import BlogListItem from './BlogListItem'
import { connect } from 'react-redux'
import { selectBlogs } from '../selectors/selectors'

const BlogList = (props) => (
    <div className="container">
        {
            props.blogs.length === 0 ? ( 
            <p className="no-blogs-message">No Blogs</p>
            ) : (
                <div className="list-body">
                    <div className="list-header">
                        <div className="show-for-mobile">Blogs</div>
                        <div className="show-for-desktop">Blog</div>
                        <div className="show-for-desktop">Reads</div>
                    </div>
                    {props.blogs.map((blog, index) => <BlogListItem key={index} {...blog}/>)}
                </div>
            )
        }
    </div>
)

const matchStateToProps = (state) => ({
    blogs: selectBlogs(state.blogs, state.filters)
})

export default connect(matchStateToProps)(BlogList)