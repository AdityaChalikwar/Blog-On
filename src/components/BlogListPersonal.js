import React from 'react'
import BlogListItem from './BlogListItem'
import { connect } from 'react-redux'
import { selectMyBlogs } from '../selectors/selectors'
 
const BlogList = (props) => (
    <div className="personal-container">
        {
            props.blogs.length === 0 ? ( 
            <p className="no-blogs-message">No Blogs</p>
            ) : (
                <div className="list-body">
                    <div className="list-header">
                        <div className="show-for-mobile">Blogs</div>
                        <div className="show-for-desktop">Blog</div>
                        <div className="show-for-desktop">Reads</div>
                    </div>
                    {props.blogs.map((blog, index) => <BlogListItem key={index} {...blog}/>)}
                </div>
            )
        }
    </div>
)

const matchStateToProps = (state) => {
    const blogs = selectMyBlogs(state.blogs)
    return {
        blogs
    }
}

export default connect(matchStateToProps)(BlogList)