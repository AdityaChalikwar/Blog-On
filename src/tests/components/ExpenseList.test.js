import {ExpensesList} from '../../components/ExpensesList'
import React from 'react'
import {shallow} from 'enzyme'
import expenses from '../fixtures/expenses'

test('should render ExpenseList with expenses', () => {
    const wrapper = shallow(<ExpensesList expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot()
})

test('should render with empty array', () => {
    const wrapper = shallow(<ExpensesList expenses={[]}/>)
    expect(wrapper).toMatchSnapshot() 
})