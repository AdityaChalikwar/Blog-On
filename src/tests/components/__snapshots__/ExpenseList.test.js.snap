// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <ExpensesListItem
    amount={15000}
    createdAt={0}
    description="rent"
    id={1}
    key="0"
    note=""
  />
  <ExpensesListItem
    amount={450000}
    createdAt={-345600000}
    description="credit card"
    id={2}
    key="1"
    note=""
  />
  <ExpensesListItem
    amount={5}
    createdAt={345600000}
    description="gum"
    id={3}
    key="2"
    note=""
  />
</div>
`;

exports[`should render with empty array 1`] = `
<div>
  <p>
    No Matches To Filters
  </p>
</div>
`;
